src/posts/2020-04-16-continuous-stuff-with-github-actions/index.md

```diff
@@ -0,0 +1,268 @@
+[github-workflow-triggers]: https://help.github.com/en/actions/reference/events-that-trigger-workflows
+[circleci-orbs]: https://circleci.com/orbs/
+[orb-awscli]: https://circleci.com/orbs/registry/orb/circleci/aws-cli#orb-source
+[actions-checkout]: https://github.com/actions/checkout/tree/v2.0.0
+[twitter]: https://twitter.com/naps62
+
+<div class="BlogPostParagraph">
+  <em>
+    This post was adapted from a talk that I gave at several local meetups over the
+    past few months. It got enough positive feedback that I felt compeled to convert it to written format. Original slides
+    are available <a href="https://slides.com/naps62/continuous-stuff" target="_blank">here</a>.
+  </em>
+</div>
+
+Last year, I took on the task of improving the continuous process over at [UTRUST][utrust]. We weren't really happy with
+the amount of work that went into our releases, and I was looking for a more agile approach, where everyone from
+developers to the QA team could do their part with low friction.
+
+Coincidentally, that project started right around the same time where GitHub Actions went into public beta. Some of our
+problems were related to shortcomings on our existing CI solution, so it felt right to see what GitHub had to offer.
+
+I ended up pleasantly surprised. But first, let's talk about what exactly was wrong with our previous CI.
+
+## Shortcomings of traditional CIs
+
+I've worked with a fair amount of CIs over the years...
+
+![CIs](./cis.png)
+
+There was always something that seemed a bit off, though.  They all do one simple, but very useful, thing: they react to
+commits
+
+The main use case for this is the now common one: to run tests for every new version of your code.  But any kind of
+automated task can be triggered, really. A deploy is very common as well, or a preview build for testing.
+
+But a lot of these automations shouldn't necessarily need a commit. They are not triggered by changes in the code.
+Deploys might be triggered by some higher-level decision, or a QA team manually approving the latest version, which was
+committed days ago.
+
+### Not everything is a commit
+
+This led to the common practice of creating "special" ways to trigger these commits. In [CircleCI][circleci], for
+example, you'd do something like this:
+
+```yaml
+workflows:
+  build:
+    release:
+      - deploy_to_staging:
+          filters:
+            branches:
+              only:
+                - staging
+```
+
+With this, commits to the `staging` branch would trigger a deploy to staging. Seems simple enough.  But now you need
+a branch for every single environment. And probably quite a few `push -f` commands, or similar git sorceries to force
+a commit from your normal workflow into a branch whose history is messier than the plot of [Primer][primer].
+
+## GitHub API to the rescue
+
+The problem here is that we're solving things the wrong way. A deploy is not a commit. It's a deploy.
```

Comment:
I always have this question: is it deploy or deployment? I thought the former was the verb and the latter the noun.

Reply:
to be honest, I don't think there's a strict definition.

we also push "commits" and not "commitments"

Reply:
Ok, that's how I've been going on about it 👍